### ISSUE TITLE
Wazuh does not show vulnerabilities related to kernel package

### ISSUE THREAD

**Issue by ibrahmgarip** [wazuh/wazuh#21980]
Hi All, 

I've installed Wazuh for my Ubuntu 22.04 servers and scanned the vulnerabilities but when I compare it to Vuls, the Wazuh doesn't show off vulnerabilities related to the kernel. 

For example, Vuls detected CVE-2023-6176 on one of my servers. When I check the description of CVE it seems valid, the vulnerability covering my kernel and Ubuntu says it will fixed in the next one. But Wazuh doesn't catch this vulnerability on my server. I've checked the database, and it shows under NVD_CVE but it doesn't listed on the VULNERABILITIES table. I've added my ossec.conf and agent.conf and also the screenshots. 

Does Wazuh not support kernel vulnerabilities or am I having a problem?

![Screenshot from 2024-02-15 15-14-57](https://github.com/wazuh/wazuh-dashboard-plugins/assets/110819566/b7e56a93-9d22-4b37-a8b3-d55227be21ed)

![Screenshot from 2024-02-15 15-03-40](https://github.com/wazuh/wazuh-dashboard-plugins/assets/110819566/f81aef18-c0b0-4b23-8ebb-859ad8c027f4)

![Screenshot from 2024-02-15 15-01-25](https://github.com/wazuh/wazuh-dashboard-plugins/assets/110819566/d7938029-3577-4348-a925-52dd69014198)

```
`
 <vulnerability-detector>
    <enabled>yes</enabled>
    <interval>24h</interval>
    <min_full_scan_interval>60h</min_full_scan_interval>
    <run_on_start>yes</run_on_start>

    <!-- Ubuntu OS vulnerabilities -->
    <provider name="canonical">
      <enabled>yes</enabled>
      <os>trusty</os>
      <os>xenial</os>
      <os>bionic</os>
      <os>focal</os>
      <os>jammy</os>
      <update_interval>1h</update_interval>
    </provider>
     
      ..........
    
   <!-- Aggregate vulnerabilities -->
    <provider name="nvd">
      <enabled>yes</enabled>
      <update_interval>1h</update_interval>
    </provider>

  </vulnerability-detector>
`
```

```
`<!-- System inventory -->
   <wodle name="syscollector">
     <disabled>no</disabled>
     <interval>1h</interval>
     <scan_on_start>yes</scan_on_start>
     <hardware>yes</hardware>
     <os>yes</os>
     <network>yes</network>
     <packages>yes</packages>
     <ports all="no">yes</ports>
     <processes>yes</processes>

     <!-- Database synchronization settings -->
     <synchronization>
       <max_eps>10</max_eps>
     </synchronization>
   </wodle>

`
```

**Comment by mhamra** [comment]
Hi Ibrahim,
In which Wazuh version are you working?
What kernel version has your Ubuntu server? (`uname -r`)
Please post the results of these requests to the Wazuh API , replacing {agent_id} with the agent number of your monitored Ubuntu server:

`GET /syscollector/{agent_id} /packages`
`GET /vulnerability/{agent_id}`

**Comment by ibrahmgarip** [comment]
Hi @mhamra, 
thank you for helping me. 
 
I've deployed another agent on Ubuntu 20.04 with 5.15.0-94-generic kernel. Wazuh agent and server versions are 4.7.2.1.
Unfortunately, I could not get any kernel-related CVE on this new agent either. I've added my API console responses below. 

[vulnerabilities.json](https://github.com/wazuh/wazuh/files/14498976/vulnerabilities.json)
[packages.json](https://github.com/wazuh/wazuh/files/14498977/packages.json)

**Comment by mhamra** [comment]
Hi @ibrahmgarip ,
In your `packages.json` file, you can find this entry showing the kernel installed in your agent `011`:

```
      {
        "scan": {
          "id": 0,
          "time": "2024-03-05T16:38:50+00:00"
        },
        "install_time": " ",
        "vendor": "Canonical Kernel Team <kernel-team@lists.ubuntu.com>",
        "section": "kernel",
        "description": "Signed kernel image generic",
        "location": " ",
        "source": "linux-signed-hwe-5.15",
        "size": 11289,
        "version": "5.15.0-94.104~20.04.1",
        "name": "linux-image-5.15.0-94-generic",
        "architecture": "amd64",
        "format": "deb",
        "priority": "optional",
        "agent_id": "011"
      }
```

This [link](https://ubuntu.com/security/CVE-2023-6176) is the vulnerability database used by the Vulnerability Detector for Ubuntu. There you can find this entry:

- "source": "linux-hwe-5.15"
- "version": "5.15.0-94.104~20.04.1"
- "name": "linux-modules-5.15.0-94-generic"

![image](https://github.com/wazuh/wazuh/assets/2949519/9f71b553-5e41-474d-a6ce-714dc6a4a2ae)

Your installed package is not vulnerable, it would be vulnerable if you have a lower version than the one marked as `Released` in Ubuntu `focal`. It is ok that the VD doesn't show the vulnerability for your server.

**Comment by rauldpm** [comment]
Closed due to inactivity, please do not hesitate to open it again or open a new issue if you need it

**Comment by leoBirebent** [comment]
Hi All,

For example, Nessus Vuls detected [CVE-2023-7042](https://ubuntu.com/security/CVE-2023-7042) on one of my servers. When I check the CVE description, it seems valid, the vulnerability covers my kernel and Ubuntu says it will be fixed by upgrading to 5.15.0-112.122. But Wazuh doesn't detect this vulnerability on my server.
![Capture d’écran du 2024-08-30 10-40-12](https://github.com/user-attachments/assets/f5c46db1-bddb-4519-8ff3-43d35ce5e456)

```
{
        "scan": {
          "id": 0,
          "time": "2024-08-30T08:26:45+00:00"
        },
        "source": "linux",
        "location": " ",
        "section": "kernel",
        "priority": "optional",
        "version": "5.15.0-105.115",
        "vendor": "Ubuntu Kernel Team <kernel-team@lists.ubuntu.com>",
        "architecture": "amd64",
        "format": "deb",
        "size": 112989,
        "description": "Linux kernel extra modules for version 5.15.0 on 64 bit x86 SMP",
        "install_time": " ",
        "name": "linux-modules-5.15.0-105-generic",
        "agent_id": "006"
      }
      ```
      Does Wazuh not support kernel vulnerabilities or am I having a problem?
